src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -420,6 +420,23 @@ private MethodSignature ParseMethodSignatureImpl(bool skipEmbeddedSignatureData)
         }
 
         public PropertySignature ParsePropertySignature()
+        {
+            try
+            {
+                _indexStack = new Stack<int>();
+                _indexStack.Push(0);
+                _embeddedSignatureDataList = new List<EmbeddedSignatureData>();
+                return ParsePropertySignatureInternal();
+            }
+            finally
```

Comment:
EcmaSignatureParser is a struct that is not meant to be reused for multiple parsing operatins. This cleanup should not be needed.

(I know that pattern is used in existing methods, but it looks unnecessary.)

Reply:
This is part of the "the modifiers stuff in the managed type system is incomprehensible". I just cargo cult my way through it.

I'll delete this part of the copy paste.

Reply:
Delete the cargo cult from the two other places in this file as well?

Reply:
I'll do this in a separate PR. I don't like deleting unrelated mystery code in a 500-line PR because the mystery can unravel itself.